src/main/java/com/spotify/docker/client/DefaultDockerClient.java

```diff
@@ -362,25 +364,72 @@ public Info info() throws DockerException, InterruptedException {
     WebTarget resource = resource()
         .path("containers").path("json");
 
+    final Map<String, List<String>> filters = newHashMap();
     for (ListContainersParam param : params) {
-      resource = resource.queryParam(param.name(), param.value());
+      if (param instanceof ListContainersFilterParam) {
+        List<String> filterValueList;
+        if (filters.containsKey(param.name())) {
+          filterValueList = filters.get(param.name());
+        } else {
+          filterValueList = Lists.newArrayList();
+        }
+        filterValueList.add(param.value());
+        filters.put(param.name(), filterValueList);
+      } else {
+        try {
+          final String encodedName = URLEncoder.encode(param.name(), UTF_8.name());
+          final String encodedValue = URLEncoder.encode(param.value(), UTF_8.name());
+          resource = resource.queryParam(encodedName, encodedValue);
+        } catch (UnsupportedEncodingException e) {
+          throw new DockerException(e);
+        }
+      }
+    }
+
+    // If filters were specified, we must put them in a JSON object and pass them using the
+    // 'filters' query param like this: filters={"dangling":["true"]}
```

Comment:
Will the parameter value always be a JSON object with a list of strings as the values? The [docs here](https://docs.docker.com/engine/reference/api/docker_remote_api_v1.18/#list-containers) imply that the exited and status filters should look like `?filters={"exited":0, "status": "running"}`, but perhaps the docs are misleading?